tests/Benchmark/BenchmarkTest.h

```diff
@@ -0,0 +1,138 @@
+    }
+
+protected:
+    TestClass m_test;
+};
+
+template <typename TestClass, typename Arg>
+class BenchmarkTestRunnerP : public ::testing::TestWithParam<Arg> {
+    static_assert(std::is_base_of<BenchmarkTestBase, TestClass>::value, "Benchmark Test should derive from ::testing::BenchmarkTestBase");
+    static_assert(std::is_constructible<TestClass, Arg>::value,
+                  "Benchmark Test Class MUST be constructible from template arguments, if any");
+
+public:
+    BenchmarkTestRunnerP() {
+        // Note: This will break when multiple args have the same type
+        m_test.reset(new TestClass(::testing::WithParamInterface<Arg>::GetParam()));
+    }
+
+protected:
+    std::unique_ptr<TestClass> m_test;
+};
+}
+}
+
+// clang-format off
+#define _BENCHMARK_CLASSNAME(test_name, test_case_name) BenchmarkInternal##test_name##test_case_name
+
+#define TEST_BENCHMARK(test_name, test_case_name, run_count)                                            \
+class _BENCHMARK_CLASSNAME(test_name, test_case_name) : public ::testing::BenchmarkTestBase   {         \
+public:                                                                                                 \
+        size_t GetRunCount() const {                                                                    \
+            return run_count;                                                                           \
+        }                                                                                               \
+        inline void Run();                                                                              \
+};                                                                                                      \
+GTEST_TEST_(test_name,                                                                                  \
+            test_case_name,                                                                             \
+            ::testing::internal::BenchmarkTestRunner<_BENCHMARK_CLASSNAME(test_name, test_case_name)>,  \
+            ::testing::internal::GetTestTypeId()) {                                                     \
+        ::testing::internal::__RunTest(m_test);                                                         \
+}                                                                                                       \
+inline void _BENCHMARK_CLASSNAME(test_name, test_case_name)::Run()
+
+#define TEST_BENCHMARK_F(test_name, test_class_name)                                                                                        \
+GTEST_TEST_(test_name, test_class_name, ::testing::internal::BenchmarkTestRunner<test_class_name>, ::testing::internal::GetTestTypeId()) {  \
+        ::testing::internal::__RunTest(m_test);                                                                                             \
+}
+
+#define BENCHMARK_REGISTER_TEST_P(test_name, test_class, generator, ...)                                                \
```

Comment:
Rationale: _they are not tests_

---
In reply to: [99247720](https://github.com/Microsoft/WinObjC/pull/1876#discussion_r99247720) [](ancestors = 99247720)